subprojects/language-groovy/src/main/java/org/gradle/api/internal/tasks/compile/SourceClassesMappingFileAccessor.java

```diff
@@ -79,7 +79,7 @@ public static void writeSourceClassesMappingFile(File mappingFile, Multimap<File
         StringBuilder sb = new StringBuilder();
 
         for (Map.Entry<File, Collection<String>> entry : mapping.asMap().entrySet()) {
```

Comment:
I think the map key should be from `String`. Looking at the `File` would require to use some working directory - which we don't have.